### PR TITLE
fix: STOP-955 handling open URL based on domain name

### PR DESCRIPTION
**Pull request by SB-venkatyadavilli**
## Addressed
[STOP-955](https://smartbear.atlassian.net/browse/STOP-955)

## Changes
Handled URL based on domain matching

## Testing
Have checked jest and manual testing.

[STOP-955]: https://smartbear.atlassian.net/browse/STOP-955?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx [hunk 1]
```diff
@@ -130,13 +130,22 @@ const externalRegex = new RegExp('^(?:[a-z]+:)?//', 'i');
...
+      const baseURL = window.location.host;
+      const hrefURL = new URL(href).host;
+
+      if (baseURL === hrefURL) {
+        // Open URL in same tab if domain match
+        return (
+          <a href={href} rel="noreferrer" title={title ? title : undefined}>
+            {children}
+          </a>
+        );
```
Comment (ben-smartbear) [inline packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx:144]: i would prefer the logic to stay in the order it is and not get inverted

also, why `noreferrer` when it's opening in the same tab?

**Review by ben-smartbear — CHANGES_REQUESTED** [review]

File: packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx (hunk 1 above)
Comment (SB-venkatyadavilli) [reply packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx]: @ben-smartbear , made changes according to review comment, could you please re-review?

File: packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx [hunk 2]
```diff
@@ -130,13 +130,27 @@ const externalRegex = new RegExp('^(?:[a-z]+:)?//', 'i');
...
+      if (baseURL === hrefURL) {
+        // Open URL in same tab if domain match
+        return (
+          <a href={href} title={title ? title : undefined}>
+            {children}
+          </a>
+        );
+      }
+      return (
+        <a href={href} target="_blank" rel="noreferrer">
```
Comment (ben-smartbear) [inline packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx:147]: why did we lose the `title` value here?

Comment (SB-venkatyadavilli) [reply packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx]: Have resolved this, could you please re-review it?

**Pull request merged**